rdtools/analysis_chains.py

```diff
@@ -242,7 +245,17 @@ def _calc_clearsky_poa(self, times=None, rescale=True, **kwargs):
             **kwargs)
         clearsky_poa = clearsky_poa['poa_global']
 
+        if aggregate:
+            interval_id = pd.Series(range(len(self.poa_global)), index=self.poa_global.index)
+            interval_id = interval_id.reindex(times).bfill()
```

Comment:
I think this would be more robust
```suggestion
            interval_id = interval_id.reindex(times.union(self.poa_global.index).bfill()
```

Here is an example that breaks the existing code:
```
original_index = pd.to_datetime([
    '2021/01/01 12:00',
    '2021/01/01 12:15:01',
    '2021/01/01 12:30:01',
    '2021/01/01 12:45:01'
])
times = pd.date_range(original_index[0], original_index[-1], freq="1min")

x = pd.Series(1, times)

interval_id = pd.Series(range(len(original_index)), index = original_index)
interval_id = interval_id.reindex(times).bfill()

out = x.groupby(interval_id).mean()
out.index = original_index
```

Reply:
Good counterexample.  What about `interval_id.reindex(times, method='backfill')` instead of `interval_id.reindex(times.union(self.poa_global.index).bfill()`?

Reply:
I like it!